### ISSUE TITLE
Hard time troubleshooting a "Corrupt Cache" issue

### ISSUE THREAD

**Issue by hpages** [Bioconductor/AnnotationHub#12]
Well, troubleshooting is not supposed to be fun anyway. However I think in this case it could be made a little bit easier. First I have to say that once I was able to figure out what documentation exactly I had to look at, the documentation was very helpful and it allowed me to quickly fix my cache corruption problem. So thanks for the great documentation!

The issue I was troubleshooting is the following cache corruption problem:
```
library(ExperimentHub)
eh <- ExperimentHub()
# snapshotDate(): 2020-02-26

bis_1072 <- eh[["EH1072"]]  # warning: this is big! (2.9G)
# see ?tissueTreg and browseVignettes('tissueTreg') for documentation
# Error: failed to load resource
#   name: EH1072
#   title: Bisulfite sequencing data from tissue Tregs (per sample)
#   reason: Corrupt Cache: resource path
#   See vignette section on corrupt cache
#   cache: /home/hpages/.cache/ExperimentHub
#   potential duplicate files: 
#     36a232eda19_1072
#     36a4e0f9ec3_1072
```

While I was troubleshooting it, I ran into a few minor rough edges that I thought it might be worth sharing here.

**1. The error message is too vague**

The recommendation to "See vignette section on corrupt cache" is too vague and it took me a while to actually find the right vignette. Naively, since I'm using ExperimentHub, I would assume that I need to look at the ExperimentHub vignette. But the "Access the ExperimentHub Web Service" vignette in ExperimentHub doesn't say anything about "corrupt cache".

My next guess was that the error message actually comes from BiocFileCache so I should probably look at the BiocFileCache vignette. (Note that I'm only able to make that guess because I know that ExperimentHub uses BiocFileCache behind the scene but most users don't know that.) No such luck: the BiocFileCache vignette doesn't say anything about "corrupt cache" either.

So I cheated (because I can, but most users won't be able to do this): I `grep`-ped the full Bioconductor code source with the `"See vignette section on corrupt cache"` pattern (I have local git clones of all software packages on my laptop) and found occurrences of the pattern in AnnotationHub code. Ah I remember now that ExperimentHub also relies on AnnotationHub so it can reuse some of the functionalities implemented there. Let's try `browseVignettes("AnnotationHub")`. Oops, it's showing a list of 4 vignettes! Not sure which one I'm supposed to look at. After opening the "Access the AnnotationHub Web Service" vignette and realizing that it says nothing about "corrupt cache" either, I finally figured out that the vignette I am supposed to look at is the **"Troubleshoot The Hubs" vignette in AnnotationHub**.

Could the error message say that?

**2. Use `bfccache(bfc)` instead of `hubCache(ah)` at the end of section 2.2.3**

I would suggest using `bfccache(bfc)` instead of `hubCache(ah)` at the end of the "2.2.3 resource path" subsection. Someone who jumps directly into that section (which is what I did) doesn't have an `ah` object around but they do have the `bfc` object and it's pointing to the right cache location (there is no such guarantee with the `ah` object, and in my case it would actually be the wrong place to remove files from).

**3. Why enforce the dplyr dialect on the reader?**

Especially when you can just do
```
bfcinfo(bfc, rid="BFC90")$rpath  # universally understood AND less typing
```
instead of
```
bfcinfo(bfc, rid="BFC90") %>% dplyr::select(rpath)  # hard-to-read dialect (except
                                                    # for dplyr fans) AND more typing
```
Furthermore, the first time I tried it, the dplyr dialect gave me an error:
```
> bfcinfo(bfc, rid="BFC90") %>% dplyr::select(rpath)
Error in bfcinfo(bfc, rid = "BFC90") %>% dplyr::select(rpath) : 
  could not find function "%>%"
```
but that's my fault for jumping directly into section 2.2.3.

Thanks!

**sessionInfo():**
```
R Under development (unstable) (2019-10-30 r77336)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /home/hpages/R/R-4.0.r77336/lib/libRblas.so
LAPACK: /home/hpages/R/R-4.0.r77336/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] dplyr_0.8.4          tissueTreg_1.7.0     ExperimentHub_1.13.5
[4] AnnotationHub_2.19.7 BiocFileCache_1.11.4 dbplyr_1.4.2        
[7] BiocGenerics_0.33.0 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3                    pillar_1.4.3                 
 [3] compiler_4.0.0                BiocManager_1.30.10          
 [5] later_1.0.0                   tools_4.0.0                  
 [7] digest_0.6.25                 bit_1.1-15.2                 
 [9] RSQLite_2.2.0                 memoise_1.1.0                
[11] tibble_2.1.3                  pkgconfig_2.0.3              
[13] rlang_0.4.4.9001              cli_2.0.1                    
[15] shiny_1.4.0                   DBI_1.1.0                    
[17] curl_4.3                      yaml_2.2.1                   
[19] fastmap_1.0.1                 httr_1.4.1                   
[21] IRanges_2.21.3                vctrs_0.2.3                  
[23] S4Vectors_0.25.12             rappdirs_0.3.1               
[25] stats4_4.0.0                  bit64_0.9-7                  
[27] tidyselect_1.0.0              Biobase_2.47.2               
[29] glue_1.3.1                    R6_2.4.1                     
[31] fansi_0.4.1                   AnnotationDbi_1.49.1         
[33] tcltk_4.0.0                   purrr_0.3.3                  
[35] blob_1.2.1                    magrittr_1.5                 
[37] promises_1.1.0                htmltools_0.4.0              
[39] assertthat_0.2.1              mime_0.9                     
[41] interactiveDisplayBase_1.25.0 xtable_1.8-4                 
[43] httpuv_1.5.2                  utf8_1.1.4                   
[45] crayon_1.3.4                  BiocVersion_3.11.1           
```

**Comment by hpages** [comment]
Today I ran into a "Corrupt Cache" problem again:
```
library(ExperimentHub)
hub <- ExperimentHub()
# snapshotDate(): 2021-01-14
fname <- hub[["EH1039"]]
# see ?TENxBrainData and browseVignettes('TENxBrainData') for documentation
# Error: failed to load resource
#   name: EH1039
#   title: Brain scRNA-seq data, 'HDF5-based 10X Genomics' format
#   reason: Corrupt Cache: resource path
#   See vignette section on corrupt cache
#   cache: /home/hpages/.cache/ExperimentHub
#   potential duplicate files: 
#     e9e175557313_1039
#     f2f2735640e7_1039
```
I would have struggled to find the "vignette section on corrupt cache" if I hadn't run into this issue before. As explained in my original report, when using **ExperimentHub** and getting an error message that tells me to consult the vignette, one cannot easily figure out that the vignette is actually in **AnnotationHub**. I would even say that finding this vignette is almost impossible for most end users. Also, since **AnnotationHub** contains several vignettes, providing the name of the vignette would help a lot. Thanks!

**Comment by lshep** [comment]
Thanks.  sorry I've had this on my todo for awhile. I'll try and look at it next week.

**Comment by lshep** [comment]
I fixed #1 and #2 points but have not changed the dplyr context.  The rest of the vignettes and the underlying BiocFileCache utilizes the dplyr syntax and format.  This is consistent.

**Comment by hpages** [comment]
Thanks.
